src/webapp/components/customize-settings-dialog/CustomizeSettingsDialog.tsx

```diff
@@ -0,0 +1,215 @@
+import { ConfirmationDialog, useLoading } from "@eyeseetea/d2-ui-components";
+import { Icon, IconButton, TextField } from "@material-ui/core";
+import React, { ChangeEvent, useCallback, useMemo, useRef, useState } from "react";
+import styled from "styled-components";
+import i18n from "../../../locales";
+import { useAppContext } from "../../contexts/app-context";
+import { getKeys } from "../../../types/utils";
+import { CustomText, CustomTextInfo, defaultCustomText } from "../../../domain/entities/CustomText";
+import MoreVertIcon from "@material-ui/icons/MoreVert";
+import _ from "lodash";
+import MenuItem from "@material-ui/core/MenuItem";
+import Menu from "@material-ui/core/Menu";
+import { useAppConfigContext } from "../../contexts/AppConfigProvider";
+import { ImportTranslationDialog, ImportTranslationRef } from "../import-translation-dialog/ImportTranslationDialog";
+import { useImportExportTranslation } from "../../hooks/useImportExportTranslation";
+
+export interface CustomizeSettingsSaveForm {
+    customText: Partial<CustomText>;
+    logo: string;
+}
+
+export interface CustomSettingsDialogProps extends CustomizeSettingsSaveForm {
+    onSave: (data: Partial<CustomizeSettingsSaveForm>) => void;
+    onClose: () => void;
+}
+
+export const CustomizeSettingsDialog: React.FC<CustomSettingsDialogProps> = props => {
+    const { onSave, customText, logo, onClose } = props;
+
+    const { usecases } = useAppContext();
+    const { exportTranslation, importTranslation } = useImportExportTranslation();
+    const loading = useLoading();
+
+    const { appCustomText } = useAppConfigContext();
+    const [logoVal, setLogo] = useState<string>(logo);
+    const [customTextVal, setCustomText] = useState<Partial<CustomText>>(customText);
+    const translationImportRef = useRef<ImportTranslationRef>(null);
+    const customTextKeys = useMemo(() => getKeys(appCustomText), [appCustomText]);
+
+    const logoHasChanges = useMemo(() => logoVal !== logo, [logoVal, logo]);
+
+    const isCustomTextDefault = useMemo(() => _.every(customText, _.isUndefined), [customText]);
+    const customTextHasChanges = useMemo(
+        () => !_.every(customTextVal, _.isUndefined) || _.isEqual(customTextVal, appCustomText),
```

Comment:
Could use the previous const `isCustomTextDefault` for the `!_.every(customTextVal, _.isUndefined) `